privacyidea/static/components/login/controllers/loginControllers.js

```diff
@@ -119,10 +119,15 @@ angular.module("privacyideaApp")
     });
 
     $scope.$on('IdleTimeout', function () {
-        console.log("Lock!");
-        $scope.logoutWarning = false;
-        $scope.$apply();
-        $scope.lock_screen();
+        if ($scope.timeout_action == "logout") {
+            console.log("Logout!");
+            $scope.logout();
+        } else if ($scope.timeout_action == "lockscreen") {
```

Comment:
Would it make more sense to actually have a default? If for any reason timeout_action is not set correctly, the screen will neither lock nor logout.
We could use an "else" instead of an "else if", so that we are secure here in any case.
What do you think?